### ISSUE TITLE
[require-top-level-describe] false positive for describe.each?

### ISSUE THREAD

**Issue by ljosberinn** [jest-community/eslint-plugin-jest#707]
tests within `describe.each` throw the `require-top-level-describe` error - is this intended?

![image](https://user-images.githubusercontent.com/29307652/98685089-3d2e9a00-2367-11eb-8f11-32ce13c6d98c.png)

**Comment by SimenB** [comment]
Nah, that's a bug. Wanna send a PR?

**Comment by ljosberinn** [comment]
I've been trying but I'm afraid I don't know enough about the internals of eslint rules.

```ts
      MemberExpression(node) {
        if (
          node.parent?.type === 'CallExpression' &&
          isDescribe(node.parent) &&
          node.type === 'MemberExpression' &&
          node.property.type === 'Identifier' &&
          node.property.name === 'each'
        ) {
          numberOfDescribeBlocks++;

          return;
        }
      },
      'MemberExpression:exit'(node: TSESTree.MemberExpression) {
        if (
          node.parent?.type === 'CallExpression' &&
          isDescribe(node.parent) &&
          node.type === 'MemberExpression' &&
          node.property.type === 'Identifier' &&
          node.property.name === 'each'
        ) {
          numberOfDescribeBlocks--;

          return;
        }
      },
```
technically that works, as in, increments `numberOfDescribeBlocks`, but I'm fairly lost beyond that.
